### PR TITLE
controller code using generated files

### PR DESCRIPTION
**Pull request by oneilcin**
I tested running the controller, it will accept KrakenCluster CRDs.  This is just a start, more work to do!

### CODE REVIEW

File: glide.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package: github.com/samsung-cnct/cluster-controller
```
Comment (alejandroEsc) [inline glide.yaml]: my only concern is that we should be using dep, or rather i thought this was what engineering agreed to.

File: main.go [hunk 2]
```diff
@@ -3,49 +3,209 @@ package main
...
+	// cluster-controller pkg libraries
+	samsungv1alpha1 "github.com/samsung-cnct/cluster-controller/pkg/apis/clustercontroller/v1alpha1"
+	clientset "github.com/samsung-cnct/cluster-controller/pkg/client/clientset/versioned"
+	samsungscheme "github.com/samsung-cnct/cluster-controller/pkg/client/clientset/versioned/scheme"
+	informers "github.com/samsung-cnct/cluster-controller/pkg/client/informers/externalversions"
+	listers "github.com/samsung-cnct/cluster-controller/pkg/client/listers/clustercontroller/v1alpha1"
+	"github.com/samsung-cnct/cluster-controller/pkg/signals"
+)
+
+const controllerAgentName = "cluster-controller"
```
Comment (alejandroEsc) [inline main.go]: good controller name.

File: main.go [hunk 3]
```diff
@@ -3,49 +3,209 @@ package main
...
+func (c *Controller) runWorker() {
+	for c.processNextWorkItem() {
+	}
+}
+
+// processNextWorkItem will read a single work item off the workqueue and
+// attempt to process it, by calling the syncHandler.
+func (c *Controller) processNextWorkItem() bool {
+	obj, shutdown := c.workqueue.Get()
+
```
Comment (alejandroEsc) [inline main.go]: remove this space here.

File: main.go [hunk 4]
```diff
@@ -3,49 +3,209 @@ package main
...
 
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the KrakenCluster resource
+// with the current status of the resource.
+func (c *Controller) syncHandler(key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
```
Comment (alejandroEsc) [inline main.go]: why isnt the error returned here?

File: main.go [hunk 5]
```diff
@@ -3,49 +3,209 @@ package main
...
+	c.recorder.Event(kc, corev1.EventTypeNormal, SuccessSynced, MessageResourceSynced)
 	return nil
 }
 
+func (c *Controller) updateKrakenClusterStatus(kc *samsungv1alpha1.KrakenCluster) error {
+	// NEVER modify objects from the store. It's a read-only, local cache.
+	// You can use DeepCopy() to make a deep copy of original object and modify this copy
+	// Or create a copy manually for better performance
+	kcCopy := kc.DeepCopy()
+	kcCopy.Status.Status = "UNKNOWN Right Now"
```
Comment (alejandroEsc) [inline main.go]: leave as UNKNOWN

File: main.go (hunk 4 above)
Comment (oneilcin) [reply main.go]: Because the error is handled here.  You can see this pattern in the sample-controller.

File: main.go (hunk 3 above)
Comment (oneilcin) [reply main.go]: ok

File: main.go (hunk 5 above)
Comment (oneilcin) [reply main.go]: ok

File: glide.yaml (hunk 1 above)
Comment (alejandroEsc) [reply glide.yaml]: Alright, ill check out what is needed to go ahead and use dep. I got my other project to work, so should be just emulating that.

**Pull request merged**